libqrexec/log.c

```diff
@@ -88,5 +88,8 @@ void setup_logging(const char *program_name) {
             errx(125, "File descriptor %d is closed, cannot continue", i);
     }
 
+    /* qrexec_logv() uses several separate fprintf calls, print them at once */
+    setlinebuf(stderr);
```

Comment:
```suggestion
    if (setvbuf(stderr, NULL, _IOLBF, 0))
        errx(1, "failed to enable line buffering on stderr");
```

`setvbuf()` is the C11 and POSIX standard interface, `setlinebuf()` is presumably a legacy alias.

Reply:
Do you think failure here really should be fatal?

Reply:
It doesn’t strictly need to be, and if there was any reasonable reason for this function to fail I would agree that it should not be fatal.  However, the only reasons I can think of are either:

1. libc is buggy.
2. libc cannot allocate an 8KiB buffer.

In both cases, something is already horribly wrong.

That said, I would not oppose changing this to a warning.

Reply:
> libc cannot allocate an 8KiB buffer.

according to documentation, the buffer is allocated only on first print, not here